tests/ignite/test_utils.py

```diff
@@ -174,6 +174,30 @@ def test_override_setup_logger(capsys):
     logging.shutdown()
 
 
+@pytest.mark.parametrize("encoding", [None, "utf-8"])
+def test_setup_logger_encoding(encoding, dirname):
+
+    fp = dirname / "log"
+    logger = setup_logger(name="logger", filepath=fp, encoding=encoding, reset=True)
+    test_words = ["hello", "你好", "привет"]
+    for w in test_words:
+        logger.info(w)
+    logging.shutdown()
+
+    with open(fp, "r") as h:
+        data = h.readlines()
+
+    if encoding == "utf-8":
+        assert len(data) == len(test_words)
+        for expected, output in zip(test_words, data):
+            assert expected in output
+    else:
+        # TODO: let's which OS will fail here
```

Comment:
Right now, only windows is failing, but both tests are failing, probably due to above `open(fp, "r")`.
https://pipelinesghubeus22.actions.githubusercontent.com/GjXvbhyRrTHRIrk9p03R8DrZ91xWSv51MZra799xsxhclrqZfD/_apis/pipelines/1/runs/30704/signedlogcontent/3?urlExpires=2024-04-23T13%3A15%3A47.3762975Z&urlSigningMethod=HMACV1&urlSignature=PBWM1r%2B3weu5dxUn2g9rvcYpPQeoqkir%2BWH%2FaM56IxU%3D